### ISSUE TITLE
Change FreeCAD (Precompiled) to .deb

### ISSUE THREAD

**Issue by chunky-milk** [Botspot/pi-apps#430]
I've precompiled FreeCAD 0.18.6 into a deb file, which will make it a lot faster and easier to install. Download link here: https://github.com/chunky-milk/freecad-rpi/releases/download/0.18.6/freecad_0.18.6-1_armhf.deb

**Comment by Botspot** [comment]
> I've precompiled FreeCAD 0.18.6 into a deb file, which will make it a lot faster and easier to install. Download link here: https://github.com/chunky-milk/freecad-rpi/releases/download/0.18.6/freecad_0.18.6-1_armhf.deb

Thank you very much! This has been needing doing.

**Comment by Botspot** [comment]
Upon uninstalling your deb, I see this:
```
Removing freecad (0.18.6-1) ...
dpkg: warning: while removing freecad, directory '/usr/local/share/mime/packages' not empty so not removed
dpkg: warning: while removing freecad, directory '/usr/local/share/icons/hicolor' not empty so not removed
dpkg: warning: while removing freecad, directory '/usr/local/share/applications' not empty so not removed
dpkg: warning: while removing freecad, directory '/usr/local/data/Mod/Material/StandardMaterial' not empty so not removed
```
Could those errors be suppressed somehow?

**Comment by Botspot** [comment]
Other than that, it worked flawlessly. FreeCAD has been updated to use your deb, while the FreeCad (precompiled) app has been deleted. And now FreeCAD no longer resides in its own category, it's alongside all the other Editors.

Commit 0b18031e021fff2d87140d273dabf8ee3d2b9502.

I'll leave this issue open for now; hopefully those "`not empty so not removed`" errors can be fixed in a later update.

**Comment by chunky-milk** [comment]
> Other than that, it worked flawlessly. FreeCAD has been updated to use your deb, while the FreeCad (precompiled) app has been deleted.  
> 
> And now FreeCAD no longer resides in its own category, it's alongside all the other Editors. Thanks again! Commit 0b18031e021fff2d87140d273dabf8ee3d2b9502.
> 
> 
> 
> I'll leave this issue open for now; hopefully those "`not empty so not removed`" errors can be fixed in a later update.

Of course! I think the error you were getting was because FreeCAD had been installed previously, so it would need to be removed to make way for the new version.

**Comment by Botspot** [comment]
> I think the error you were getting was because FreeCAD had been installed previously, so it would need to be removed to make way for the new version.

No, FreeCAD was not installed prior to this. Can you attempt to repeat the error?

1. Install the new FreeCAD app.
2. Uninstall FreeCAD.
3. See the error.

**Comment by chunky-milk** [comment]
@Botspot I got this:
![image](https://user-images.githubusercontent.com/71036629/110256370-9352c900-7f66-11eb-9f54-d14b70e6df37.png)

**Comment by Botspot** [comment]
> @Botspot I got this:

In that case, it would be awesome if you figured out how to hide those errors from a deb-creation standpoint.

After all, [not all debs complain about /usr/local/bin being not empty](https://github.com/Botspot/pi-apps/issues/276#issuecomment-771188124)!

**Comment by chunky-milk** [comment]
> In that case, it would be awesome if you figured out how to hide those errors from a deb-creation standpoint.
> 
> 
> 
> After all, [not all debs complain about /usr/local/bin being not empty](https://github.com/Botspot/pi-apps/issues/276#issuecomment-771188124)!

I'll work on it.

**Comment by chunky-milk** [comment]
> Upon uninstalling your deb, I see this:
> 
> ```
> 
> Removing freecad (0.18.6-1) ...
> 
> dpkg: warning: while removing freecad, directory '/usr/local/share/mime/packages' not empty so not removed
> 
> dpkg: warning: while removing freecad, directory '/usr/local/share/icons/hicolor' not empty so not removed
> 
> dpkg: warning: while removing freecad, directory '/usr/local/share/applications' not empty so not removed
> 
> dpkg: warning: while removing freecad, directory '/usr/local/data/Mod/Material/StandardMaterial' not empty so not removed
> 
> ```
> 
> Could those errors be suppressed somehow?

Just curious, but after these messages and attempting to remove the package from the system, is FreeCAD still present?

**Comment by Botspot** [comment]
> Just curious, but after these messages and attempting to remove the package from the system, is FreeCAD still present?

No, the package was removed successfully and completely. As far as I know, those errors are purely cosmetic or perhaps indicate a messed up deb configuration.

**Comment by Itai-Nelken** [comment]
> Upon uninstalling your deb, I see this:
> 
> ```
> Removing freecad (0.18.6-1) ...
> dpkg: warning: while removing freecad, directory '/usr/local/share/mime/packages' not empty so not removed
> dpkg: warning: while removing freecad, directory '/usr/local/share/icons/hicolor' not empty so not removed
> dpkg: warning: while removing freecad, directory '/usr/local/share/applications' not empty so not removed
> dpkg: warning: while removing freecad, directory '/usr/local/data/Mod/Material/StandardMaterial' not empty so not removed
> ```
> 
> Could those errors be suppressed somehow?

that happens for some weird reason when uninstalling some "homemade" debs, it isn't anything serious and I even saw it happen in some apt packages.

**Comment by chunky-milk** [comment]
@Botspot Should this issue be closed now ?

**Comment by Botspot** [comment]
> @Botspot Should this issue be closed now ?

Sure, if you have given up on hiding those errors.

**Comment by Itai-Nelken** [comment]
I asked if there is a way to supress those errors in the unix stackexchange: https://unix.stackexchange.com/questions/638440/how-to-supress-remove-dpkg-warning-while-removing-x-directory-x-y-z-not-em
while researching this I got to another stackexchange question/answer that says that this error is caused by the deb putting files in `/usr/local/` (that is correct with this deb I'm pretty sure and with my QEMU and box86 debs). the solution would be to put the files in `/usr/` instead, but the warning is harmless.

**Comment by Botspot** [comment]
Okay interesting.
